Sources/Kitura/RouterRequest.swift

```diff
@@ -102,13 +93,11 @@ public class RouterRequest {
     /// The currently matched section of the URL.
     public internal(set) var matchedPath = ""
 
-    /// The original URL as a string.
-    public var originalURL: String {
-        return serverRequest.urlString
-    }
+    /// The URL from the request if properly received
+    public internal(set) var url : URL?
 
-    /// The URL.
-    public let url: String
+    /// The URL from the request as URLComponents
+    public internal(set) var urlComponents = URLComponents()
```

Comment:
That sounds good to me. 

So to summarize - add `urlComponents: URLComponents`, deprecate `url: String` and `parsedURL: URLParser` in favor of urlComponents, and remove internal use of the deprecated variables.

And do we want to use the same approach in Kitura-net as well?

@shmuelk @ianpartridge : Could you confirm this sounds ok?